demo/api/src/auth/access-control.service.ts

```diff
@@ -1,20 +1,8 @@
-import {
-    AccessControlServiceInterface,
-    ContentScope,
-    ContentScopesForUser,
-    CurrentUserInterface,
-    PermissionsForUser,
-    User,
-    UserPermissions,
-} from "@comet/cms-api";
+import { AccessControlService as CometAccessControlService, ContentScopesForUser, PermissionsForUser, User, UserPermissions } from "@comet/cms-api";
 import { Injectable } from "@nestjs/common";
 
 @Injectable()
-export class AccessControlService implements AccessControlServiceInterface {
-    canAccessScope(requestScope: ContentScope, user: CurrentUserInterface): boolean {
-        if (!user.domains) return true; //all domains
-        return user.domains.includes(requestScope.domain);
-    }
+export class AccessControlService extends CometAccessControlService {
```

Comment:
Do you think it's good idea to use inheritance here? Can't this be done via composition somehow?

Reply:
Yes, I think it is a good idea. Inheritance is not bad by default and i think here it is exactly what we want to do.

Reply:
If you really want to avoid it we could move the can move the default [canAccessScope](https://github.com/vivid-planet/comet/pull/1358/files#diff-39e97b1d273b5c27eadbda60864d01be59a8503feaa972c25e68238db9848039R9) to another location, maybe making it non overwritable at all.

Reply:
If not, then I'm fine with using inheritance here

Reply:
> You mean that accessControlService in the module instantiation is mandatory?

Yes. If we tend to override it in the application code all the time anyways, why add a fallback/default value at all?

Reply:
> Will getPermissionsForUser/getContentScopesForUser be required for a typical application?

Yes, I think so. Mainly for allowing access to a certain group of users (e.g. admins) without to need to assign them manually.

Reply:
:+1:

how is the group of a user (typically) stored?

Reply:
Depends on the IDP and what is delivered with the ID-Token. But the simplest solution would be based on the domain of the e-mail address.

Reply:
Ok, that will work in our setup and will be used more or less everywhere.

But that is the only thing we can do(?)

Reply:
Here in the demo I could add some data to the static users to simulate data delivered from the IDP. But I don't think this gives a benefit in terms of understanding the logic.

Reply:
I made the AccessControlService mandatory and renamed the base class to AbstractAccessControlService: dee1c4dea3ed22b882042250efb104d3b93e6377